src/legacy/core_plugins/data/public/search/search_bar/components/saved_query_management/saved_query_management_component.tsx

```diff
@@ -61,19 +60,44 @@ export const SavedQueryManagementComponent: FunctionComponent<Props> = ({
 }) => {
   const [isOpen, setIsOpen] = useState(false);
   const [savedQueries, setSavedQueries] = useState([] as SavedQuery[]);
+  const [count, setTotalCount] = useState(0);
   const [activePage, setActivePage] = useState(0);
 
+  const fetchSavedQueries = async () => {
+    const savedQueryItems = await savedQueryService.findSavedQueries(
+      undefined,
+      perPage,
+      activePage + 1
+    );
+    const sortedSavedQueryItems = sortBy(savedQueryItems, 'attributes.title');
+    setSavedQueries(sortedSavedQueryItems);
+  };
+
+  const fetchSavedQueryCount = async () => {
+    const savedQueryCount = await savedQueryService.getSavedQueryCount();
+    setTotalCount(savedQueryCount);
+  };
+
   useEffect(() => {
-    const fetchQueries = async () => {
-      const allSavedQueries = await savedQueryService.getAllSavedQueries();
-      const sortedAllSavedQueries = sortBy(allSavedQueries, 'attributes.title');
-      setSavedQueries(sortedAllSavedQueries);
+    // Only fetch the total count on initial render.
```

Comment:
I went back and forth on fetching the total count on each page change and decided not to add the additional network call.

Reply:
The comment above is no longer valid. https://github.com/elastic/kibana/pull/47646#discussion_r334089305